### PR TITLE
codemirror file view: Improve line selection UX

### PR DESCRIPTION
**Pull request by fkling**
This PR makes two changes:

To achieve parity with the current (HTML-based) blob view, the background color of the line numbers of selected lines also changes.

Right now "select-by-drag" doesn't update the selected lines when the mouse cursor leaves the line number gutter during the process. This was changed by binding the corresponding event handler to `window` instead.

Before/after video:

https://user-images.githubusercontent.com/179026/183857589-9782b79f-6c97-4428-9692-4da4ba5d8bb7.mp4

## Test plan

Click and drag to select lines. Line decoration includes line number gutter and moving the mouse outside the line number gutter still updates the selected lines in real time.

## App preview:

- [Web](https://sg-web-fkling-cm-fv-sl.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-yzozjfbwyd.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/repo/blob/codemirror/linenumbers.ts [hunk 1]
```diff
@@ -8,7 +17,10 @@ import { EditorView, Decoration, lineNumbers, ViewPlugin, PluginValue, ViewUpdat
  */
 export type SelectedLineRange = { line: number; endLine?: number } | null
 
-const highlighedLineDecoration = Decoration.line({ class: 'selected-line' })
+const selectedLineDecoration = Decoration.line({ class: 'selected-line' })
+const selectedLineGutterMarker = new (class extends GutterMarker {
+    elementClass = 'selected-line'
+})()
```
Comment (umpox) [inline client/web/src/repo/blob/codemirror/linenumbers.ts:23]: beautiful 😉

File: client/web/src/repo/blob/codemirror/linenumbers.ts [hunk 2]
```diff
@@ -163,22 +192,23 @@ export function selectableLineNumbers(config: {
...
-                    if (dragging) {
-                        const newEndline = view.state.doc.lineAt(line.from).number
-                        const { endLine } = view.state.field(selectedLines) ?? {}
-                        if (endLine !== newEndline) {
-                            view.dispatch({
-                                effects: setEndLine.of(newEndline),
-                                annotations: lineSelectionSource.of('gutter'),
-                            })
+
+                    function onmousemove(event: MouseEvent) {
```
Comment (umpox) [inline client/web/src/repo/blob/codemirror/linenumbers.ts:196]:
```suggestion
                    function onmousemove(event: MouseEvent): void {
```

**Review by umpox — APPROVED** [review]
Nice!

File: client/web/src/repo/blob/codemirror/linenumbers.ts (hunk 2 above)
Comment (fkling) [reply client/web/src/repo/blob/codemirror/linenumbers.ts:196]: Ugh, I always forget to run the linter :D Thanks!

File: client/web/src/repo/blob/codemirror/linenumbers.ts (hunk 1 above)
Comment (Bill-gates6969) [reply client/web/src/repo/blob/codemirror/linenumbers.ts]: yummy

**Pull request merged**